orm/query.go

```diff
@@ -709,9 +710,8 @@ loop:
 		if ind != -1 {
 			field := order[:ind]
 			sort := order[ind+1:]
-			switch internal.UpperString(sort) {
-			case "ASC", "DESC", "ASC NULLS FIRST", "DESC NULLS FIRST",
-				"ASC NULLS LAST", "DESC NULLS LAST":
+			reg := regexp.MustCompile(`(?i)^((ASC|DESC)( NULLS FIRST| NULLS LAST)?)|(COLLATE [a-zA-Z1-9_]+ (ASC|DESC))$`)
```

Comment:
Now, I don't work with `COLLATE` much but I assume that a `COLLATE` can still specify `NULLS FIRST/LAST` whereas this regex will only allow `NULLS FIRST/LAST` or `COLLATE` at a time.

Also, in the Postgres docs, I see that collations are quoted as identifiers, so probably it's proper to extract the `COLLATE` name as well and specify it as `types.Ident`.

I might play with Postgres later to verify my first question.

Reply:
Same here @dsonck92, I updated my PR to support the case you mentioned.

Reply:
Meanwhile I was so close to a fix already (though it bypasses the regex so I may share it for sake of posterity to let @vmihailenco decide which is preferred). But yes, the code is now covering more features which I think is best. AFAICS, LGTM